infra/compile.py

```diff
@@ -640,7 +663,13 @@ def compile_expr(tree, ctx):
         e = compile_expr(tree.elt, ctx2)
         if e != arg:
             if "await" in e:
-                out = "(await Promise.all(" + out + ".map(async (" + arg + ") => " + e + ")))"
+                try:
+                    t = find_hint(gen.iter, "type")
+                    out = 'Object.keys(' + out + ')'
+                    assert t == "map"
+                except:
```

Comment:
Kind ugly to do a general-purpose catch here, but OK

Reply:
Probably we just need a default parameter to `find_hint`?

Reply:
Great idea - done.